src/pages/patientView/trialMatch/TrialMatchTable.tsx

```diff
@@ -0,0 +1,354 @@
+} from "../../../shared/model/MatchMiner";
+import styles from './style/trialMatch.module.scss';
+import { computed } from "mobx";
+import LazyMobXTable from "../../../shared/components/lazyMobXTable/LazyMobXTable";
+import SampleManager from "../sampleManager";
+import DefaultTooltip, { placeArrowBottomLeft } from "../../../shared/components/defaultTooltip/DefaultTooltip";
+
+export type ITrialMatchProps = {
+    sampleManager?: SampleManager | null;
+    trials: Array<ITrial>;
+    trialMatches:Array<ITrialMatch>;
+    nctTrials:Array<INctTrial>;
+    containerWidth: number;
+}
+
+export type ITrialMatchState = {
+    detailedTrialMatches: Array<IDetailedTrialMatch>;
+}
+
+enum ColumnKey {
+    ID = 'ID',
+    TITLE = 'Title',
+    INTERVENTIONS = 'Interventions',
+    MATCHINGCRITERIA = 'Matching Criteria'
+}
+
+class TrialMatchTableComponent extends LazyMobXTable<IDetailedTrialMatch> {
+
+}
+
+@observer
+export default class TrialMatchTable extends React.Component<ITrialMatchProps, ITrialMatchState> {
+
+    constructor(props: ITrialMatchProps) {
+        super(props);
+        if (props.sampleManager) {
+            this.state = { detailedTrialMatches: this.buildDetailedTrialMatches(props.trialMatches, props.trials, props.nctTrials, props.sampleManager) };
+        } else {
+            this.state = { detailedTrialMatches: this.buildDetailedTrialMatches(props.trialMatches, props.trials, props.nctTrials) };
+        }
+    }
+
+    @computed
+    get columnsWidth() {
+        return {
+            [ColumnKey.ID]: 0.1 * this.props.containerWidth,
+            [ColumnKey.TITLE]: 0.24 * this.props.containerWidth,
+            [ColumnKey.INTERVENTIONS]: 0.16 * this.props.containerWidth,
+            [ColumnKey.MATCHINGCRITERIA]: 0.5 * this.props.containerWidth
+        };
+    }
+
+    private _columns = [{
+        name: ColumnKey.ID,
+        render: (trial: IDetailedTrialMatch) => (
+            <div>
+                <span><a target="_blank" href={"https://www.mskcc.org/cancer-care/clinical-trials/" + trial.protocolNo}>{trial.protocolNo}</a></span><br/>
+                <span><a target="_blank" href={"https://clinicaltrials.gov/ct2/show/" + trial.nctId}>{trial.nctId}</a></span><br/>
+                <span>{trial.status}</span>
+            </div>
+        ),
+        sortBy:(trial: IDetailedTrialMatch) => (trial.protocolNo),
+        // download: (trial: IDetailedTrialMatch) => (trial.protocolNo + ', ' + trial.nctId + ', ' + trial.status),
+        width: this.columnsWidth[ColumnKey.ID]
+    }, {
+        name: ColumnKey.TITLE,
+        render: (trial: IDetailedTrialMatch) => (
+                <span>{trial.shortTitle}</span>
+        ),
+        sortBy:(trial: IDetailedTrialMatch) => (trial.shortTitle),
+        // download:(trial: IDetailedTrialMatch) => (trial.shortTitle),
+        width: this.columnsWidth[ColumnKey.TITLE]
+    }, {
+        name: ColumnKey.MATCHINGCRITERIA,
+        render: (trial: IDetailedTrialMatch) => {
```

Comment:
This render method seems embedded lots of logics. Probably better abstract them out and properly tested.

Reply:
```
<div>
      {armMatch.matches.map((clinicalGroupMatch: any, cgIndex:number) => (
             <span className={styles.criteriaContainer}>
                   {this.getGenomicMatch(clinicalGroupMatch)}
                   {this.getClinicalMatch(clinicalGroupMatch)}
                   <If condition={cgIndex < armMatch.matches.length - 1}><hr className={styles.criteriaHr}/></If>
            </span>
       ))}
 </div>
```